module/VuFind/src/VuFind/Related/Bookplate.php

```diff
@@ -0,0 +1,242 @@
+ *
+ * @category VuFind
+ * @package  Related_Records
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:related_records_modules Wiki
+ */
+class Bookplate implements RelatedInterface
+{
+    /**
+     * Bookplate config
+     */
+    protected $config;
+
+    /**
+     * Data fields (usually Solr)
+     */
+    protected $fields;
+
+    /**
+     * Bookplate strings
+     */
+    protected $bookplateStrs;
+
+    /**
+     * Bookplate image names or full URLs
+     */
+    protected $bookplateImages;
+
+    /**
+     * Bookplate thumbnail image names or thumbnail URLs
+     */
+    protected $bookplateThumbnails;
+
+    /**
+     * URL template for full bookplate
+     */
+    protected $fullUrlTemplate;
+
+    /**
+     * URL temlate for thumbnail
+     */
+    protected $thumbUrlTemplate;
+
+    /**
+     * Display bookplate titles?
+     */
+    protected $displayTitles;
+
+    /**
+     * Constructor
+     *
+     * @param VuFind\Config\PluginManager $configLoader PluginManager
+     */
+    public function __construct(\VuFind\Config\PluginManager $configLoader)
+    {
+        $this->configLoader = $configLoader;
+        $this->config = $this->configLoader->get('Related');
+    }
+
+    /**
+     * Establishes base settings for bookplates.
+     *
+     * @param string                            $settings Settings from config.ini
+     * @param \VuFind\RecordDriver\AbstractBase $driver   Record driver object
+     *
+     * @return void
+     */
+    public function init($settings, $driver)
+    {
+        $this->fields = $driver->getRawData();
+        $this->bookplateStrs = $this->getBookplateData('titles');
+        $this->bookplateImages = $this->getBookplateData('imgFull');
+        $this->bookplateThumbnails = $this->getBookplateData('imgThumb');
+        $this->fullUrlTemplate = $this->getBookplateFullUrlTemplate();
+        $this->thumbUrlTemplate = $this->getBookplateThumbUrlTemplate();
+        $this->displayTitles = $this->displayBookplateTitles();
+    }
+
+    /**
+     * Get an array of data representing bookplates.
+     *
+     * @param $s string name of data to retrieve.
+     *
+     * @return array
+     */
+    protected function getBookplateData($s)
+    {
+        $data = [
+            'titles' => $this->getBookplateTitlesField(),
+            'imgThumb' => $this->getBookplateThumbnailsField(),
+            'imgFull' => $this->getBookplateFullImagesField(),
+        ];
+        $field = $data[$s];
+        if (!empty($field) && isset($this->fields[$field])) {
+            return $this->fields[$field];
+        }
+        return [];
+    }
+
+    /**
+     * Get the full bookplate URL string template.
+     *
+     * @return string
+     */
+    protected function getBookplateFullUrlTemplate()
+    {
+        return isset($this->config->Bookplate->bookplate_full) ?
```

Comment:
All of these isset checks can be simplified using the null coalescing operator. If you run `vendor/bin/phing php-cs-fixer` it should auto-fix them for you, assuming you have the latest Composer dependencies installed (php-cs-fixer recently improved its ability to auto-correct these issues).

Reply:
Beautiful. This is much nicer!